### ISSUE TITLE
Add support for annotated type parameters

### ISSUE THREAD

**Issue by Emmeral** [victools/jsonschema-generator#287]
Given a class
```java

public class Dummy {
  public Map<String, @Max(10) Integer> map;
}
```
which is annotated using jakarta bean validation annotations.
The schema generated currently using the jakarta module is
```json
{
  "type":"object",
  "properties":{
    "map":{
      "type":"object",
      "additionalProperties":{
        "type":"integer"
      }
    }
  }
}
```

But the following schema would better conform to the desired intention:
```json
{
  "type":"object",
  "properties":{
    "map":{
      "type":"object",
      "additionalProperties":{
        "type":"integer",
        "maximum":10
      }
    }
  }
}
```

A similar behavior would also be desirable for other collections (e.g. `List<@Max(10) Integer`).

**Comment by CarstenWickner** [comment]
I agree that supporting this would be beneficial.
Right now, I don't know how to access these annotations properly.
I'm open to suggestions (and pull requests). 😉

**Comment by CarstenWickner** [comment]
These annotations are supported for container types already via the [`getContainerItemAnnotation()`](https://github.com/victools/jsonschema-generator/blob/a43b22822db982c42182398be9ebad34ce682849/jsonschema-generator/src/main/java/com/github/victools/jsonschema/generator/FieldScope.java#L139-L147) method.
One would have to adjust the approach somehow to apply it to the additional attributes.

I haven't come up with a clean solution yet though.

**Comment by CarstenWickner** [comment]
HI @Emmeral,

I'm happy to let you know, that I found a way to do this in the standard `Option.MAP_VALUES_AS_ADDITIONAL_PROPERTIES`.

It'll be available with the next release `4.28.0` then.
